internal/datatransfer/monitor.go

```diff
@@ -135,13 +133,13 @@ func (m *Monitor) HasStorageDefined() bool {
 	return storage.S3 != nil
 }
 
-func (m *Monitor) SetStorage(fs FileSync) {
+func (m *Monitor) SetStorage(fs model.FileSync) {
 	m.syncMutex.Lock()
 	defer m.syncMutex.Unlock()
 	m.fsSync = fs
 }
 
-func (m *Monitor) getFsSync() (FileSync, error) {
+func (m *Monitor) getFsSync() (model.FileSync, error) {
```

Comment:
This method needs to be refactored, at lead from lines 183 to 201 can be extracted (the for loop on the workloads)

Reply:
will check

Reply:
Refactored a bit, what do you think now?

Reply:
Better but why not extracting the `for` loop on the workload into another method (lines 184 to 194)?

Reply:
Ok, good by me; I know I tend to create a lot a small methods every time I have something in an `if` or a `for`/`while` loop but we all have different way of coding :)

Reply:
Thanks for your understanding and for your feedback. In my opinion, the code looks much better after your suggestions than it did before it :smile: